Automattic/sensei
# Send email using GB template replacing placeholders to teacher on course start

Fixes https://github.com/Automattic/sensei/issues/6477

### Changes proposed in this Pull Request

* Added functionality to send emails from templates
* Added functionality to replace placeholders in templates
* Added email generator for sending an email to teacher when student starts course

### Testing instructions
- Run composer install
- Enable the email post type following the direction here https://github.com/Automattic/sensei/pull/6487
- Create a teacher user with an email you can check
- Create a course with the aforementioned teacher as the author
- Enroll in that course as a student
- A mail should be sent, or you can go to Student Management tab of that course and reset the progress to resend the email
- Go to Sensei->Settings->Emails->Teacher Emails
- Check if an email template is created for this email

### New Actions
`sensei_send_html_email` - Send templated email using email template name and replacement data placeholders.

### New Filters
`sensei_email_replacements` - Filter the array of replacement values to be used for replacing the placeholders in email

`sensei_email_styles` - Filter the styles to be added in the email inline

`sensei_email_allowed_stylesheets` - Filter the list of allowed stylesheets that are already existing

`sensei_email_stylesheets` - Filter the stylesheets to add/remove new or existing stylesheets

### Screenshot / Video
## Editor
![image](https://user-images.githubusercontent.com/6820724/218098604-a81b6c6e-3319-46b7-acb9-843c5d99b965.png)

## Email in client after replacing
![image](https://user-images.githubusercontent.com/6820724/218098766-50a0b947-2d61-4f52-8e78-081a198abd77.png)
